packages/shared/src/hooks/useShareOrCopyLink.ts

```diff
@@ -0,0 +1,30 @@
+interface UseShareOrCopyLinkProps {
+  link: string;
+  text: string;
+  trackEvent?: () => void;
+  copyLink?: () => Promise<void>;
+}
+export function useShareOrCopyLink({
+  link,
+  text,
+  trackEvent,
+  copyLink,
+}: UseShareOrCopyLinkProps): () => Promise<void> {
+  const onShareOrCopy = async () => {
+    trackEvent?.();
```

Comment:
I think it's better to track the event here with an actual function call to the real `trackEvent` and receive a function that generates the event instead of doing the call as well

Reply:
I am not 100% sure what it means.

Reply:
@idoshamun I moved both track events inside the hooks. Let me know if this works for you.